articles/anomaly-detection/passwords-checking.md

```diff
@@ -0,0 +1,229 @@
+- If you think  a token might have been compromised blacklist it immediately.
+- Delete the input data and results once finished.
+
+## Usage
+This tool is made available through our [Management API](/api/management/v2). It has an endpoint that allows you to check a pre-defined list of users and passwords, which must be provided as a csv file, against a database connection. Each password will be checkend in background and some time later you will be able to get the results, the errors, and finally delete the information from Auth0 service.
+
+### Preparing the input file
+In order to run this job you need to upload a list of users and passwords to check. This list must be provided as a CSV (coma separated value) where each row must has te following format:
+```
+{username},{email},{password}
+```
+
+#### Format considerations:
+- Must not has header row
+- Each row *must* have a maximum of 400 characters
+- If any of the values contain a coma, the whole value must be quoted
+- Email and username are optional, but you must provide one of them (and not both)
+
+The following JSON schema describes a row and might be used to validate it before uploading:
+```js
+   {
+     type: 'object',
+     anyOf: [
+       {
+         properties: {
+           email: {
+             type: 'string',
+             minLength: 1,
+             maxLength: 100,
+             description: 'Email to check'
+           },
+           plain_text_password: {
+             type: 'string',
+             minLength: 1,
+             maxLength: 100,
+             description: 'Password to check'
+           }
+         },
+         required: ['plain_text_password', 'email'],
+         additionalProperties: false
+       },
+       {
+         properties: {
+           username: {
+            type: 'string',
+            minLength: 1,
+            maxLength: 100,
+            description: 'Username to check'
+           },
+           plain_text_password: {
+             type: 'string',
+             minLength: 1,
+             maxLength: 100,
+             description: 'Password to check'
+           }
+         },
+         required: ['plain_text_password', 'username'],
+         additionalProperties: false
+       }
+     ]
+   }
+```
+
+##### File Example
+```
+john.doe0@example.com,,secret
+,john.doe2,secret
+john.doe3@example.com,,"secret, really secret"
+,john.doe2,secr et
+```
+
+These rows are all valid input rows. Note the quotation around "secret, really secret"
+because that password contains a coma. Whitespace does not need quotation as stated on
+4th line.
+
+### Starting the job
+Once you have the file ready, you are ready to start a job to check the file against your database connection. To do so you need to execute a `POST` request against `/api/passwords-checking` with encoding type `multipart/form-data`.
+
+Your request should contain the following parameters:
+
+- passwords (the csv file, detailed above, that you are uploading, which contains data to check)
```

Comment:
actually I was explicitly banned from using the passwords_file name. To be consistent with users import which has a field `users` for the file